### ISSUE TITLE
Pythonnet version: 2.4.0.dev0  running:  import clr, throws exception System.NotImplementedException: No support for x86

### ISSUE THREAD

**Issue by vanillycpp** [pythonnet/pythonnet#772]
### Environment

-   Pythonnet version: 2.4.0.dev0
-   Python version: 2.7.1
-   Operating System: Win7 x86
-  .NET version 4.6.01055

### Details

on installation of 2.3.0 vrsion have this error
https://github.com/pythonnet/pythonnet/issues/609

I installed pythonnet 2.4.0.dev0 from current master
and try to run
```
import clr
```

and got this error
```
Unhandled Exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.NotImplementedException: No support for x86
   at Python.Runtime.TypeManager.NativeCode.get_Active()
   at Python.Runtime.TypeManager.InitializeNativeCodePage()
   at Python.Runtime.TypeManager.InitializeSlots(IntPtr type, Type impl)
   at Python.Runtime.TypeManager.CreateMetaType(Type impl)
   at Python.Runtime.MetaType.Initialize()
   at Python.Runtime.Runtime.Initialize()
   at Python.Runtime.PythonEngine.Initialize(IEnumerable`1 args, Boolean setSysArgv)
   at Python.Runtime.PythonEngine.InitExt()
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.RuntimeType.InvokeMember(String name, BindingFlags bindingFlags, Binder binder, Object target, Object[] providedArgs, ParameterModifier[] modifiers, CultureInfo culture, String[] namedParams)
   at clrModule.PyInit_clr()
```

**Comment by denfromufa** [comment]
How did you install or build pythonnet?

**Comment by vanillycpp** [comment]
I run commands from "GitHub installation from master branch"
https://github.com/pythonnet/pythonnet/wiki/Installation
but I had troubles with this command
```
setup.py bdist_wheel --xplat
```

I output with error
```
running bdist_wheel
running build
running build_ext
error: [WinError 2] The system cannot find the file specified
```

then I try to look at setyp.py and found that in this part
https://github.com/pythonnet/pythonnet/blob/f8e147b97112c526c6548e900dee0fc41cdb39b1/setup.py#L311-L321

here `use_shell = DEVTOOLS == "Mono" or DEVTOOLS == "dotnet"` 
i had no .net sdk installed and no Mono, just Visual Studio
and my DEVTOOLS had value MsDev15

and wthout shell (shell=false) this command
`subprocess.check_call(cmd, shell=use_shell)`
got error `error: [WinError 2] The system cannot find the file specified`

when I installed [.net sdk](https://www.microsoft.com/net/download/thank-you/dotnet-sdk-2.1.403-windows-x86-installer)
have no error more
I think it should be somewehre in requirenments, or I shoud see error to with problem definition

**Comment by denfromufa** [comment]
@vanillycpp the option ` --xplat` is for .net core build. So try without this option, since you are using .NET Framework.

**Comment by vanillycpp** [comment]
installing without `--xplat` has no errors during installation
but when I try to `import clr` I got error for witch I startetd this issue

**Comment by denfromufa** [comment]
@vanillycpp what do you get from `import platform; platform.machine()` ?

Currently the recognized machine mappings are:

```csharp
static readonly Dictionary<string, MachineType> MachineTypeMapping = new Dictionary<string, MachineType>()
        {
            { "i386", MachineType.i386 },
            { "x86_64", MachineType.x86_64 },
            { "amd64", MachineType.x86_64 },
        };
```

**Comment by vanillycpp** [comment]
this code `import platform; platform.machine()`
gives me `x86`

from cmd `systeminfo`
```
OS Version:                        6.1.7601 Service Pack 1 build 7601
System type:                      X86-based PC
```

**Comment by denfromufa** [comment]
@vanillycpp ok, we need to add `x86` to the Dictionary above, thanks for reporting this. let me know if you would like to make this PR. Here are all the possible values for `platform.machine()`:

https://stackoverflow.com/questions/45125516/possible-values-for-uname-m

**Comment by filmor** [comment]
From this overview alone, we need to add `x86`, `i686`, and `x64`.

**Comment by filmor** [comment]
I've added the names, after the PR is merged this should work fine. Odd that this didn't show up in AppVeyor...

**Comment by filmor** [comment]
Fixed by #775.

**Comment by asferatum** [comment]
Hellow!
I installed pythonnet from pythonnet‑2.5.2‑cp39‑cp39‑win_amd64.whl from [here ](https://www.lfd.uci.edu/~gohlke/pythonlibs/#pythonnet) and so I have the same error as in head of topic. 

Next code 
`import platform; platform.machine()`
returns `'AMD64'` in upper case, so I think this is same problem.
